components/participant_table/commons/participant_table_base.lua

```diff
@@ -0,0 +1,399 @@
+---@field entries ParticipantTableEntry
+
+---@param args table
+function ParticipantTable:readSection(args)
+	local config = self.readConfig(args, self.config)
+	local section = {config = config}
+
+	local keys = Array.filter(Array.extractKeys(args), function(key)
+		return String.contains(key, '^%d+$') or
+			String.contains(key, '^p%d+$') or
+			String.contains(key, '^player%d+$')
+	end)
+
+	local entriesByName = {}
+
+	--need custom sort so it doesn't compare number with string
+	local sortKeys = function(tbl, key1, key2) return tostring(key1) < tostring(key2) end
+
+	for _, key in Table.iter.spairs(keys, sortKeys) do
+		local entry = self:readEntry(args, key, config)
+		if entriesByName[entry.name] then
+			error('Duplicate Input "|' .. key .. '=' .. args[key] .. '"')
+		end
+
+		entriesByName[entry.name] = entry
+	end
+
+	section.entries = Array.map(Import.importFromMatchGroupSpec(config, entriesByName), function(entry)
+		entry.opponent = Opponent.resolve(entry.opponent, config.resolveDate, {syncPlayer = config.syncPlayers})
+		return entry
+	end)
+
+	Array.sortInPlaceBy(section.entries, function(entry) return entry.name:lower() end)
+
+	table.insert(self.sections, section)
+end
+
+---@class ParticipantTableEntry
+---@field opponent standardOpponent
+---@field name string
+---@field note string?
+---@field dq boolean
+
+---@param sectionArgs table
+---@param key string|number
+---@param config any
+---@return ParticipantTableEntry
+function ParticipantTable:readEntry(sectionArgs, key, config)
+	--if not a json assume it is a solo opponent
+	local opponentArgs = Json.parseIfTable(sectionArgs[key]) or Logic.isNumeric(key) and {
+		type = Opponent.solo,
+		name = sectionArgs[key],
+	} or {
+		type = Opponent.solo,
+		name = sectionArgs[key],
+		link = sectionArgs[key .. 'link'],
+		flag = sectionArgs[key .. 'flag'],
+		team = sectionArgs[key .. 'team'],
+	}
+
+	assert(Opponent.isType(opponentArgs.type) and opponentArgs.type ~= Opponent.team,
+		'Missing or unsupported opponent type for "' .. sectionArgs[key] .. '"')
+
+	local opponent = Opponent.readOpponentArgs(opponentArgs)
+
+	if config.sortPlayers and opponent.players then
+		table.sort(opponent.players, function (player1, player2)
+			local name1 = (player1.displayName or player1.name):lower()
+			local name2 = (player2.displayName or player2.name):lower()
+			return name1 < name2
+		end)
+	end
+
+	return {
+		dq = Logic.readBool(opponentArgs.dq or sectionArgs[key .. 'dq']),
+		note = opponentArgs.note or sectionArgs[key .. 'note'],
+		opponent = opponent,
+		name = Opponent.toName(opponent),
+	}
+end
+
+---@return self
+function ParticipantTable:store()
+	if self.config.noStorage then return self end
+
+	local lpdbTournamentData = {
+		tournament = Variables.varDefault('tournament_name'),
+		parent = Variables.varDefault('tournament_parent'),
+		series = Variables.varDefault('tournament_series'),
+		shortname = Variables.varDefault('tournament_tickername'),
```

Comment:
iirc i copied it from ppt
can double check tomorrow when on pc

Reply:
copied from ppt
https://github.com/Liquipedia/Lua-Modules/blob/f755fe3e94705769fd99736baffc3660e419aef3/components/prize_pool/commons/prize_pool_base.lua#L801

Reply:
i think the idea is to mostly retrieve it from lpdb_tournament and not from lpdb_placement
as for why it is what it is in ppt iirc most wikis had it like that (for whatever reason) in the old ppts

Reply:
Since we've marked it as deprecated, it means we previously identified it as not-needed. Likely it's backwards compatibility, I think it was common the put in the tickername before.

Reply:
```suggestion
```